pywisp/utils.py

```diff
@@ -747,10 +747,72 @@ def updateData(self):
         self.setText(self.widgetName + '\n' + self.valueOn)
 
 
-class MovableSlider(QSlider, MovableWidget):
+class DoubleSlider(QSlider):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        super().setMinimum(0)
+        super().setMaximum(1)
+
+        self._minValue = 0.0
+        self._maxValue = 1.0
+
+        self._stepSize = 1
+        self._invStepSize = 1
+
+    @property
+    def _valueRange(self):
+        vRange = self._maxValue - self._minValue
+        self._invStepSize = vRange / self._stepSize
+
+        return vRange
+
+    def value(self):
+        curValue = float(super().value())
+        return curValue / self._invStepSize * self._valueRange + self._minValue
+
+    def setValue(self, value):
+        super().setValue(int((value - self._minValue) / self._valueRange * self._invStepSize))
+
+    def setMinimum(self, value):
+        if value > self._maxValue:
+            raise ValueError("Minimum limit cannot be higher than maximum")
+
+        self._minValue = value
+        super().setMinimum(int((self._minValue - self._minValue) / self._valueRange * self._invStepSize))
+        super().setMaximum(int((self._maxValue - self._minValue) / self._valueRange * self._invStepSize))
+
+    def setMaximum(self, value):
+        if value < self._minValue:
+            raise ValueError("Minimum limit cannot be higher than maximum")
+
+        self._maxValue = value
+        super().setMinimum(int((self._minValue - self._minValue) / self._valueRange * self._invStepSize))
+        super().setMaximum(int((self._maxValue - self._minValue) / self._valueRange * self._invStepSize))
+
+    def setTickInterval(self, p_int):
+        self._stepSize = p_int
+
+        vRange = self._maxValue - self._minValue
+        self._invStepSize = vRange / self._stepSize
+
+        super().setMinimum(int((self._minValue - self._minValue) / self._valueRange * self._invStepSize))
```

Comment:
and again.

might be helpful to just define a conversion function that you then use?